staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -787,3 +799,20 @@ func AuthorizeClientBearerToken(loopback *restclient.Config, authn *Authenticati
 	tokenAuthorizer := authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup)
 	authz.Authorizer = authorizerunion.New(tokenAuthorizer, authz.Authorizer)
 }
+
+func newAPIServerID() string {
+	hostname, err := os.Hostname()
```

Comment:
I'm not sure we want to disclose the host name. Why not just use a new UUID?

Reply:
It helps debugging features built with the ID, without getting behind the load balancer and querying each apiserver in an HA cluster. That said, I like the idea in https://github.com/kubernetes/kubernetes/pull/95533#discussion_r517687292. Probably we can avoid folding debug info into the ID. 

BTW kube-controller-manager discloses the host name in its leader election record: https://github.com/kubernetes/kubernetes/blob/f75316e294c4c1ccc4afc0d89c78bbde43859c0d/cmd/kube-controller-manager/app/controllermanager.go#L263

Reply:
Having the host name may come handy if an apiserver gets restarted and one wants to debug features built with the ID (I'm thinking of priority and fairness, maybe?).

I think we can drop the host name and pid, and just use a new UUID if you want. Adding those data in future will be backwards compatible.

Reply:
Yeah, let's start with just a UUID, sorry.

Reply:
done